src/control/server/config.go

```diff
@@ -497,36 +497,23 @@ func validateMultiServerConfig(log logging.Logger, c *Configuration) error {
 		}
 		seenValues[mountConfig] = idx
 
-		if len(scmConf.DeviceList) > 0 {
-			devStr := fmt.Sprintf("scm_list:%s", sliceToString(scmConf.DeviceList))
-
-			if seenIn, exists := seenValues[devStr]; exists {
-				log.Debugf("%s in %d duplicates %d", devStr, idx, seenIn)
+		for _, dev := range scmConf.DeviceList {
+			if seenIn, exists := seenScmSet[dev]; exists {
+				log.Debugf("scm device %s in %d duplicates %d", dev, idx, seenIn)
 				return FaultConfigDuplicateScmDeviceList(idx, seenIn)
 			}
-			seenValues[devStr] = idx
+			seenScmSet[dev] = idx
 		}
 
 		bdevConf := srv.Storage.Bdev
-		if len(bdevConf.DeviceList) > 0 {
-			devStr := fmt.Sprintf("bdev_list:%s", sliceToString(bdevConf.DeviceList))
-
-			if seenIn, exists := seenValues[devStr]; exists {
-				log.Debugf("%s in %d duplicates %d", devStr, idx, seenIn)
+		for _, dev := range bdevConf.DeviceList {
+			if seenIn, exists := seenBdevSet[dev]; exists {
+				log.Debugf("bdev device %s in %d duplicates %d", dev, idx, seenIn)
 				return FaultConfigDuplicateBdevDeviceList(idx, seenIn)
```

Comment:
nit: Duplicate->Overlapping

Reply:
Yep, spotted this after pushing previously. Updated.